java/sidecar-common/src/main/java/com/instaclustr/sidecar/jackson/MapBackedTypeIdResolver.java

```diff
@@ -28,29 +29,26 @@ public String idFromValue(final Object value) {
 
     @Override
     public String idFromValueAndType(final Object value, final Class<?> suggestedType) {
-        return typeMappings.inverse().get(suggestedType).toString();
+        return typeMappings.inverse().get(suggestedType);
     }
 
-    @Override
-    public JavaType typeFromId(final DatabindContext context, final String id) {
-
-        if (id == null) {
-            return null;
-        }
-
-        final OperationType operationType = typeMappings.keySet().stream().filter(type -> type.toString().toLowerCase().equals(id.toLowerCase())).findFirst().orElse(null);
+    public Class<? extends T> classFromId(final String id) {
+        return typeMappings.get(id);
+    }
 
-        if (operationType == null) {
-            return null;
-        }
+    public Set<String> typeIds() {
```

Comment:
I need this for know what types we have in param converter and its subsequent exception thrown from there